x-pack/plugins/observability/server/services/slo/tasks/summary_cleanup_task.ts

```diff
@@ -0,0 +1,212 @@
+              'slo.id': sloId,
+            },
+          },
+          {
+            term: {
+              'slo.revision': Number(revision),
+            },
+          },
+        ],
+      },
+    };
+  });
+};
+
+export class SloSummaryCleanupTask {
+  private abortController = new AbortController();
+  private logger: Logger;
+  private taskManager?: TaskManagerStartContract;
+  private soClient?: SavedObjectsClientContract;
+  private esClient?: ElasticsearchClient;
+
+  constructor(taskManager: TaskManagerSetupContract, logger: Logger) {
+    this.logger = logger;
+
+    taskManager.registerTaskDefinitions({
+      [TASK_TYPE]: {
+        title: 'SLO Definitions Cleanup Task',
+        timeout: '3m',
+        maxAttempts: 1,
+        createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+          return {
+            run: async () => {
+              return this.runTask();
+            },
+
+            cancel: async () => {
+              this.abortController.abort('[SLO] Definitions clean up Task timed out');
+            },
+          };
+        },
+      },
+    });
+  }
+
+  runTask = async () => {
+    const runAt = new Date().toISOString();
+
+    if (this.soClient && this.esClient) {
+      const finder = this.soClient.createPointInTimeFinder<Pick<StoredSLO, 'id' | 'revision'>>({
+        type: SO_SLO_TYPE,
+        perPage: 1000,
+        fields: ['id', 'revision'],
+      });
+      let searchAfterKey: SortResults | undefined;
+      let soIdsToCheck: string[] = [];
+      let sloSummaryIdsToDelete: string[] = [];
+
+      for await (const response of finder.find()) {
+        const soItems = response.saved_objects.map(
+          (so) => `${so.attributes.id}${SEPARATOR}${so.attributes.revision}`
+        );
+
+        const { sloSummaryIds, searchAfter } = await this.fetchSloSummariesIds(searchAfterKey);
+        searchAfterKey = searchAfter;
+
+        const { missingSummaryIds, missingSOIds } = findMissingItems(
+          soItems.concat(soIdsToCheck),
+          sloSummaryIds.concat(sloSummaryIdsToDelete)
+        );
+
+        soIdsToCheck = missingSOIds;
+        sloSummaryIdsToDelete = missingSummaryIds;
+        if (searchAfterKey === undefined) {
+          break;
+        }
+      }
+
+      await finder.close();
+
+      if (sloSummaryIdsToDelete.length > 0) {
+        this.logger.info(
+          `[SLO] Deleting ${sloSummaryIdsToDelete.length} SLO Summaries from the summary index`
+        );
+        this.logger.info(JSON.stringify(sloSummaryIdsToDelete));
+
+        await this.esClient.deleteByQuery({
+          index: SLO_SUMMARY_DESTINATION_INDEX_PATTERN,
+          query: {
+            bool: {
+              should: getDeleteQueryFilter(sloSummaryIdsToDelete.sort()),
+            },
+          },
+        });
+      }
+    }
+
+    return { state: { lastRunAt: runAt } };
+  };
+
+  fetchSloSummariesIds = async (searchAfter?: SortResults) => {
+    if (this.esClient) {
```

Comment:
🍰  I would check for the undefined esClient and return early, then you can remove the need for the brackets for the main flow.

```suggestion
    if (!this.esClient) {
        return {
            searchAfter: undefined,
            sloSummaryIds: [] as string[],
          };
      }
      
      // main logic
```